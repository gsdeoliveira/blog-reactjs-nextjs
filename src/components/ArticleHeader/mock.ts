import { ArticleHeaderProps } from '.';
import { mappedData as data } from '../../api/map-data';

const { title, excerpt, cover, author, categories, createdAt } = data.posts[0];

export default {
  title,
  excerpt,
  cover,
  author,
  categories,
  createdAt,
} as ArticleHeaderProps;
