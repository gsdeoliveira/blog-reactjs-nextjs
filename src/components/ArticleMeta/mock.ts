import { ArticleMetaProps } from '.';

export default {
  createdAt: '2023-09-27T11:26:43.234Z',
  author: {
    data: {
      id: '1',
      attributes: {
        displayName: 'Gabriel Sousa',
        slug: 'gabriel-sousa',
      },
    },
  },
  categories: [
    {
      id: '1',
      attributes: {
        displayName: 'JavaScript',
        slug: 'java-script',
      },
    },
    {
      id: '2',
      attributes: {
        displayName: 'React',
        slug: 'react',
      },
    },
  ],
} as ArticleMetaProps;
