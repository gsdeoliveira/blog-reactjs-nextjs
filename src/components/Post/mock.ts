import { PostProps } from '.';
import { data } from '../../api/dados.json';

const {
  title,
  excerpt,
  cover,
  content,
  author,
  categorias: { categories },
  createdAt,
} = data.posts.data[0].attributes;

export default {
  title,
  excerpt,
  cover,
  content,
  author,
  categories,
  createdAt,
} as PostProps;
