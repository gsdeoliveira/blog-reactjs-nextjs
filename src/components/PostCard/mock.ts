import { data as dataJson } from '../../api/dados.json';

// export default dataJson.posts.data[0].attributes;

export default {
  id: dataJson.posts.data[0].id,
  ...dataJson.posts.data[0].attributes,
};
