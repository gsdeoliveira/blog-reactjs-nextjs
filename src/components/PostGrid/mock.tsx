import { data } from '../../api/dados.json';

export default {
  posts: data.posts.data.map((post) => ({
    id: post.id,
    title: post.attributes.title,
    cover: post.attributes.cover,
    excerpt: post.attributes.excerpt,
    slug: post.attributes.slug,
  })),
};
