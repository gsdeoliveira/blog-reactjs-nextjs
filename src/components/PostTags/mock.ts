import { data } from '../../api/dados.json';

export default { tags: data.posts.data[0].attributes.tags.data };
