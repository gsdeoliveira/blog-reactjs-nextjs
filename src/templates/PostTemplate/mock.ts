import { PostTemplateProps } from '.';
import { mappedData as data } from '../../api/map-data';

export default {
  settings: data.setting,
  post: data.posts[0],
} as PostTemplateProps;
