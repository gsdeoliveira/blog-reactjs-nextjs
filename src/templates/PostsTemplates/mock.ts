import { PostsTemplateProps } from '.';
import { data } from '../../api/dados.json';

const posts = data.posts.data.map((post) => {
  return {
    id: post.id,
    title: post.attributes.title,
    excerpt: post.attributes.excerpt,
    cover: post.attributes.cover,
    content: post.attributes.content,
    tags: post.attributes.tags.data,
    slug: post.attributes.slug,
    createdAt: post.attributes.createdAt,
  };
});
export default {
  settings: data.setting,
  posts: posts,
} as PostsTemplateProps;
