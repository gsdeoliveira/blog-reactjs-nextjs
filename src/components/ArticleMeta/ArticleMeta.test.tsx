import { screen } from '@testing-library/react';
import { renderTheme } from '../../../styles/render-theme';
import { ArticleMeta, ArticleMetaProps } from '.';

import mock from './mock';

const props: ArticleMetaProps = mock;

describe('<ArticleMeta />', () => {
  it('should render author and category links', () => {
    renderTheme(<ArticleMeta {...props} />);

    expect(screen.getByRole('link', { name: 'Gabriel Sousa' })).toHaveAttribute(
      'href',
      '/author/gabriel-sousa',
    );
    expect(screen.getByRole('link', { name: 'JavaScript' })).toHaveAttribute(
      'href',
      '/category/java-script',
    );
    expect(screen.getByRole('link', { name: 'React' })).toHaveAttribute(
      'href',
      '/category/react',
    );
  });

  it('should format date', () => {
    renderTheme(<ArticleMeta {...props} />);

    expect(screen.getByText('27 de set. de 2023')).toHaveAttribute(
      'datetime',
      props.createdAt,
    );
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<ArticleMeta {...props} />);
    expect(container).toMatchSnapshot();
  });
});
