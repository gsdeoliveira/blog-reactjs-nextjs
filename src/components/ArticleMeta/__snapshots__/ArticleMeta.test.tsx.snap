// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleMeta /> should match snapshot 1`] = `
.c0 {
  font-size: 1.6rem;
  color: #AAAAAA;
  font-style: italic;
}

.c0 .categories span::after {
  content: ', ';
}

.c0 .categories span:last-child::after {
  content: '';
}

.c0 a {
  color: #dc143c;
  text-decoration: none;
  transition: all 300ms ease-in-out;
}

.c0 a:hover {
  filter: brightness(50%);
}

<div>
  <div
    class="c0"
  >
    <p>
      <span>
        Por 
      </span>
      <a
        href="/author/gabriel-sousa"
      >
        Gabriel Sousa
      </a>
      <span
        class="separator"
      >
         | 
      </span>
      <time
        datetime="2023-09-27T11:26:43.234Z"
      >
        27 de set. de 2023
      </time>
      <span
        class="separator"
      >
         | 
      </span>
      <span
        class="categories"
      >
        <span>
          <a
            href="/category/java-script"
          >
            JavaScript
          </a>
        </span>
        <span>
          <a
            href="/category/react"
          >
            React
          </a>
        </span>
      </span>
    </p>
  </div>
</div>
`;
